osu.Game/Screens/Play/Player.cs

```diff
@@ -362,7 +363,10 @@ private void onBreakTimeChanged(ValueChangedEvent<bool> isBreakTime)
 
         private void updateOverlayActivationMode()
         {
-            bool canTriggerOverlays = DrawableRuleset.IsPaused.Value || !gameplayOverlaysDisabled.Value;
+            if (!this.IsCurrentScreen())
+                return;
```

Comment:
why is this check required?

Reply:
This check has been added to prevent `OverlayActivationMode` from being updated when the `Player` isn't the current screen (could be triggered previously by changing the value of the setting while in the player loader, according to this comment: https://github.com/ppy/osu/pull/9766#pullrequestreview-463824776)

Reply:
if so then I think I would prefer deferring setting up the subscriptions until `OnEntering()` as proposed above.